awx/main/fields.py

```diff
@@ -878,9 +879,12 @@ def validate_template_string(type_, key, tmpl):
 
         def validate_extra_vars(key, node):
             if isinstance(node, dict):
-                return {validate_extra_vars(key, k): validate_extra_vars("{key}.{k}".format(key=key, k=k), v) for k, v in node.items()}
+                for k, v in node.items():
+                    validate_template_string("extra_vars", 'a key' if key is None else key, k)
```

Comment:
I changed the call to only allow strings as keys, so no `{['foo']: 'bar'}` kind of syntax. Reason is because I can't get JSON to accept that data structure in the first place, so there would be no way to send it to begin with.